SilvermanAGutData from miaTime, resolve errors related to microbiomeDataSets

TuomasBorman opened the pull request:
We did not have a dataset with reference sequences. I added SilvermanAGutDataSet from microbiomeDataSets

TuomasBorman committed:
Add SilvermanAGutData

TuomasBorman committed:
sync

Sync

# Conflicts:
#	NEWS
#	R/mia.R
#	man/mia-datasets.Rd

TuomasBorman committed:
up

TuomasBorman commented on tests/testthat/test-5getExperimentCrossAssociation.R:
```diff
@@ -3,8 +3,25 @@ context("getExperimentCrossAssociation")
     
-    # Get data
-    mae <- microbiomeDataSets::peerj32()
+    # Try 5 times to fetch the data
+    for(i in seq_len(5) ){
+        mae <- tryCatch(
+            {
+                # Try to fetch the data 
+                microbiomeDataSets::peerj32()
+            },
+            error = function(cond) {
+                # If it was not possible to fetch the data, give FALSE
+                return(NULL)
+            }
+        )
+        # Break if mae has the data
+        if( !is.null(mae) ){
+            break
+        }
+    }
+    # Run tests if the data fetch was successful
+    if( !is.null(mae) ){
     ############################### Test input ###############################
```
Tests used peerj32 data and some test reference values are hard-coded. I changed this little bit. --> it is tested 5 times if the data can be fetched from microbiomeDataSets. If it was successful, tests are run. otherwise they are skipped

antagomir commented on R/mia.R:
```diff
@@ -202,3 +196,37 @@ NULL
+#' and daily to capture sub-daily dynamics of microbial community originating 
+#' from human feces. The data consists of 413 taxa from 639 samples. 
+#' The data set can be used to investigate longitudinal dynamics of microbial 
+#' community in a controlled environment. 
```
This is very nice time series data. Hence it would better with the miaTime package. Could we put it there instead?

TuomasBorman replied:
Sure, I can make a PR and just copy-paste these

TuomasBorman committed:
up

TuomasBorman committed:
up

TuomasBorman committed:
up

Sync

# Conflicts:
#	DESCRIPTION
#	tests/testthat/test-5getExperimentCrossAssociation.R

TuomasBorman committed:
up

TuomasBorman committed:
up

TuomasBorman committed:
up

TuomasBorman committed:
sync

Sync

# Conflicts:
#	DESCRIPTION

TuomasBorman commented:
Good to go(?)

antagomir commented:
Is this with time series? Just thinking if that would be better placed to miaTime?

TuomasBorman commented:
Actually, it is in miaTime. --> mia uses it in tests, so it is loaded from miaTime

antagomir commented:
perfect. good to go for me at least.
